src/test/kotlin/com/example/passmanager/repositories/PassOwnerRepositoryImplTest.kt

```diff
@@ -2,99 +2,97 @@ package com.example.passmanager.repositories
 
 import com.example.passmanager.domain.MongoPassOwner
 import com.example.passmanager.testcontainers.WithMongoTestContainer
-import com.example.passmanager.util.OptimisticLockTestUtils.getOptimisticLocksAmount
 import com.example.passmanager.util.PassOwnerFixture.getOwnerWithUniqueFields
-import com.example.passmanager.util.PassOwnerFixture.passOwnerToCreate
 import org.assertj.core.api.Assertions.assertThat
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.aggregation.Fields
 import org.springframework.data.mongodb.core.exists
 import org.springframework.data.mongodb.core.findById
 import org.springframework.data.mongodb.core.query.Criteria.where
 import org.springframework.data.mongodb.core.query.Query.query
 import org.springframework.data.mongodb.core.query.isEqualTo
+import reactor.kotlin.test.test
 import kotlin.test.Test
-import kotlin.test.assertEquals
 import kotlin.test.assertFalse
 
 @SpringBootTest
 @WithMongoTestContainer
-class PassOwnerRepositoryImplTest {
+internal class PassOwnerRepositoryImplTest {
     @Autowired
-    private lateinit var mongoTemplate: MongoTemplate
+    private lateinit var mongoTemplate: ReactiveMongoTemplate
 
     @Autowired
     private lateinit var passOwnerRepository: PassOwnerRepository
 
     @Test
     fun `finding pass by existing id should return pass owner by id`() {
         // GIVEN
-        val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
+        val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields()).block()
 
         // WHEN
-        val passOwnerById = passOwnerRepository.findById(insertedPassOwnerId.toString())
+        val passOwnerById = passOwnerRepository.findById(insertedPassOwner!!.id.toString())
 
         // THEN
-        assertThat(passOwnerById?.id).isNotNull().isEqualTo(insertedPassOwnerId)
+        passOwnerById.test()
+            .assertNext { assertThat(it).isEqualTo(insertedPassOwner) }
+            .verifyComplete()
     }
 
     @Test
     fun `inserting pass owner in collection should return created pass owner`() {
-        // WHEN
+        // GIVEN // WHEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
 
         // THEN
-        assertThat(mongoTemplate.findById<MongoPassOwner>(insertedPassOwnerId!!)).isEqualTo(insertedPassOwner)
+        insertedPassOwner
+            .test()
+            .assertNext { ownerIdAfterInsert ->
```

Comment:
`inserting pass owner in collection should return created pass owner` - the name of the test.

So, assert:
1. the result of the called function
2. find the record in the DB and assert it separately